viadot/utils.py

```diff
@@ -201,12 +201,21 @@ def get_sql_server_table_dtypes(
 
 
 def _cast_df_cols(df):
```

Comment:
Let's add a param to specify which types should be casted. This is needed because your issue was with `object` columns, but your fix also casts `bool` and other types; this should not be the case. I think you only need to cast cols of type `object`, so eg. `_cast_df_cols(df=df, types=["object"])`. 

Also, your change impacts parts of code already using this util in a way that's hard to test (eg. `gen_bulk_insert_query_from_df()` -- this way you can also avoid that.

Reply:
[✨ Added ability to select types to convert](https://github.com/dyvenia/viadot/pull/701/commits/3d8aeef136a34e581958dbd045a9dfb9ca26cda3)